components/domains/renewalDomainsBox.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { FunctionComponent, useEffect, useState } from "react";
+import styles from "../../styles/components/registerV2.module.css";
+import { Checkbox, Skeleton, Tooltip } from "@mui/material";
+import InfoIcon from "../UI/iconsComponents/icons/infoIcon";
+import { isStarkRootDomain } from "../../utils/stringService";
+import { hexToDecimal } from "../../utils/feltService";
+import { useAccount } from "@starknet-react/core";
+
+type RenewalDomainsBoxProps = {
+  helperText: string;
+  setSelectedDomains: React.Dispatch<
+    React.SetStateAction<Record<string, boolean> | undefined>
+  >;
+  selectedDomains?: Record<string, boolean>;
+};
+
+const RenewalDomainsBox: FunctionComponent<RenewalDomainsBoxProps> = ({
+  helperText,
+  setSelectedDomains,
+  selectedDomains,
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [ownedDomains, setOwnedDomains] = useState<FullId[]>([]);
+  const { address } = useAccount();
+
+  useEffect(() => {
+    if (address) {
+      fetch(
+        `${
+          process.env.NEXT_PUBLIC_SERVER_LINK
+        }/addr_to_full_ids?addr=${hexToDecimal(address)}`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          const ownedDomainsToSet: FullId[] = data.full_ids.filter(
+            (identity: FullId) => isStarkRootDomain(identity?.domain)
+          );
+          setOwnedDomains(ownedDomainsToSet);
+          setSelectedDomains(
+            ownedDomainsToSet.reduce((acc, identity) => {
+              acc[identity.domain] = true; // Initially set all to true. Adjust as needed.
+              return acc;
+            }, {})
+          );
+          setIsLoading(false);
+        });
+    }
+  }, [address]);
+
+  const handleCheckboxChange = (domain: string) => {
+    setSelectedDomains((prevState) => ({
+      ...prevState,
+      [domain]: !prevState?.[domain],
+    }));
+  };
+
+  return isLoading ? (
```

Comment:
When the user is not connected, the domain box is always in loading mode, it's not really clear for the user that he's not connected